sql/pgwire/v3.go

```diff
@@ -622,8 +622,8 @@ func (c *v3Conn) sendResponse(resp sql.Response, formatCodes []formatCode, sendD
 		return c.sendCommandComplete(nil)
 	}
 	for _, result := range resp.Results {
-		if result.PErr != nil {
-			if err := c.sendError(result.PErr.String()); err != nil {
+		if result.Err != nil {
+			if err := c.sendError(result.Err.Error()); err != nil {
```

Comment:
I still think it's a good idea because that error will be very simple, and SQL cares about very little of what a roachpb.Error holds. The distinction between user-facing txn and kv-facing txn are subtle and pgwire deals in the user-facing one. Better to limit roachpb.Error to interactions with KV and to handle all that complexity in one place (Executor).